tools/tplgtool2.py

```diff
@@ -882,14 +882,23 @@ def arg_split(value: str) -> "list[str]":
         "Split comma-delimited arguements value."
         return [v.strip() for v in value.split(',')]
 
+    def glob_or_abs(rootpath: Path, filename: str) -> Path:
+        # this is a workround for Path.glob(): it didn't implement for absolute path,
+        # even though the document said the path can be absolute:
+        # https://docs.python.org/3.8/library/glob.html#glob.glob
```

Comment:
You're confusing https://docs.python.org/3.8/library/glob.html#glob.glob with https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob

Reply:
This works:
```python
[ x for x in pathlib.Path("/").glob("usr/*/ls*") ]
```
This does not work:
```python
[ x for x in pathlib.Path("/").glob("/usr/*/ls*") ]

 NotImplementedError("Non-relative patterns are unsupported")
```
I presume the latter is what you ran into, confirm?

If I guessed correctly then the question is why do you need to look at `"/"` + `"/usr/*/ls*"`? How about not getting in this situation in the first place?

Reply:
@marc-hb Thanks for your correction. We have many ways to give topology paths, for example:
* `tplgtool2.py -t /lib/firmware/intel/sof-tplg sof-tgl-nocodec.tplg`: find `/lib/firmware/intel/sof-tplg/sof-tgl-nocodec.tplg` 
* `tplgtool2.py sof-tgl-nocodec.tplg`: find `sof-tgl-nocodec.tplg` in current working directory
* `tplgtool2.py /lib/firmware/intel/sof-tplg/sof-tgl-nocodec.tplg`: absolute path
* `tplgtool2.py -t /lib/firmware/intel/sof-tplg sof-tgl-nocodec.tplg,sof-tgl-rt5682.tplg`: find `/lib/firmware/intel/sof-tplg/sof-tgl-nocodec.tplg`  and `/lib/firmware/intel/sof-tplg/sof-tgl-rt5682.tplg`
* `tplgtool2.py -t /lib/firmware/intel/sof-tplg sof-tgl-*.tplg`: find all files match `sof-tgl-*.tplg` in `/lib/firmware/intel/sof-tplg`
*  and so on...

So this is not for searching `/usr/*/ls` in `/`, but for searching `/usr/*/ls` when root path is unspecified.

Most of these are designed to keep compatible with the legacy tool:
<https://github.com/thesofproject/sof-test/blob/530d8eb24abdf69794733c75658681082be7bd51/tools/tplgtool.py#L881-L900>
The 'wild-card' style filename could be a replacement for the special filename `all` in legacy code.

Or maybe you have a better design?